docs/tutorials/creating-your-first-substrate-chain/test.md

```diff
@@ -0,0 +1,187 @@
+
+```js
+import React, { useState, useEffect } from "react";
+import { Form, Input, Grid, Message } from "semantic-ui-react";
+import { blake2AsHex } from "@polkadot/util-crypto";
+
+import { useSubstrate } from "../substrate-lib";
+import { TxButton } from "../substrate-lib/components";
+
+export default function ProofOfExistence(props) {
+  const { api } = useSubstrate();
+  const { accountPair } = props;
+  const [status, setStatus] = useState("");
+  const [digest, setDigest] = useState("");
+  const [owner, setOwner] = useState("");
+  const [block, setBlock] = useState(0);
+
+  let fileReader;
+
+  const bufferToDigest = () => {
+    const content = Array.from(new Uint8Array(fileReader.result))
+      .map(b => b.toString(16).padStart(2, "0"))
+      .join("");
+
+    const hash = blake2AsHex(content, 256);
+    setDigest(hash);
+  };
+
+  const handleFileChosen = file => {
+    fileReader = new FileReader();
+    fileReader.onloadend = bufferToDigest;
+    fileReader.readAsArrayBuffer(file);
+  };
+
+  useEffect(() => {
+    let unsubscribe;
+
+    api.query.templateModule
+      .proofs(digest, result => {
+        setOwner(result[0].toString());
+        setBlock(result[1].toNumber());
+      })
+      .then(unsub => {
+        unsubscribe = unsub;
+      });
+
+    return () => unsubscribe && unsubscribe();
+  }, [digest, api.query.templateModule]);
+
+  function isClaimed() {
+    return block !== 0;
+  }
+
+  return (
+    <Grid.Column>
+      <h1>Proof Of Existence</h1>
+      <Form success={!!digest && !isClaimed()} warning={isClaimed()}>
+        <Form.Field>
+          <Input
+            type="file"
+            id="file"
+            label="Your File"
+            onChange={e => handleFileChosen(e.target.files[0])}
+          />
+          <Message success header="File Digest Unclaimed" content={digest} />
+          <Message
+            warning
+            header="File Digest Claimed"
+            list={[digest, `Owner: ${owner}`, `Block: ${block}`]}
+          />
+        </Form.Field>
+
+        <Form.Field>
+          <TxButton
+            accountPair={accountPair}
+            label={"Create Claim"}
+            setStatus={setStatus}
+            type="TRANSACTION"
+            attrs={{ params: [digest], tx: api.tx.templateModule.createClaim }}
+            disabled={isClaimed() || !digest}
+          />
+          <TxButton
+            accountPair={accountPair}
+            label="Revoke Claim"
+            setStatus={setStatus}
+            type="TRANSACTION"
+            attrs={{ params: [digest], tx: api.tx.templateModule.revokeClaim }}
+            disabled={!isClaimed() || owner !== accountPair.address}
+          />
+        </Form.Field>
+        <div style={{ overflowWrap: "break-word" }}>{status}</div>
+      </Form>
+    </Grid.Column>
+  );
+}
+```
+
+We won't walk you step by step through the creation of this component, but do
+look over the code comments to learn what each part is doing.
```

Comment:
there are a total of zero code comments in that file. Either we do add comments or remove this sentence ;) .

Reply:
yeah, something i meant to do but missed. We def want to comment it